docs/contributor/adr/004-layered-architecture.md

```diff
@@ -0,0 +1,79 @@
+# ADR 004 - Adoption of Layered Architecture (3-Tier)
+
+## Status
+
+Accepted
+
+## Context
+
+We need to design a maintainable, testable, and scalable architecture for lifecycle-manager, which interacts with Kubernetes APIs and includes business logic and orchestration of resources. A clear layering and therefore clear responsibilities of code components are missing and need to be introduced, which will also ease navigating the code base.
+
+## Decision
```

Comment:
Should we add some naming conventions as decisions? e.g: Controller should named as `XXXController`, Service should be `XXXService`, Repository should be `XXXRepository`? I saw you have examples using such conventions already in this ADR https://github.com/kyma-project/lifecycle-manager/pull/2429/files#diff-1df6f7ea20316af79ccac20b787d5d73187dad8a545d9634a832195148392fcdR34, maybe it's better we define it as rule.

Reply:
Ben and myself had exactly the same thought and agreed to have it discussed in one of the next arch rounds 👍🏻